### ISSUE TITLE
ResultCount always returns 0

### ISSUE THREAD

**Issue by CallMeBruce** [jrentenaar/Office-365-Extractor#6]
Hi, part way through running this (excellent) script over several runs to extract various logs, it stopped working as expected.

On investigation it appears that the ResultCount property is always 0 (even when there are results returned). 

The only further information I have about it is that it was working and then during a run it started to generate "Transport errors" (as if network was down?). I restarted the script and though it worked, it now didn't save any data (even for Record Types that had just successfully been queried and exported). I stepped through the code and then found that the ResultCount was always 0, which breaks lots of stuff.

Any light you can shed on this would be appreciated, I have tried using a different user account and from different network sources, all same result.

If I can provide more detail to assist please let me know.

**Comment by jrentenaar** [comment]
Hi, thanks! This is a new error for me, and a strange one. 

The script works by checking if there are any logs between for example 12:00 and 13:00. If the ResultCount =0, this means there are no logs between 12:00 and 13:00 and it moves on to the next time interval. If there are logs found the count is higher then 1 and the logs will be written to a CSV file. 

Could you try to run option 1 of the script "Show available log sources and amount of logging". This shows the available logging and how many log events there are available. If the script does not find any log events, then I would recommend you to check if the account you are using has the right permissions: _You have to be assigned the View-Only Audit Logs or Audit Logs role in Exchange Online to search the Office 365 audit log._

And are you sure Audit Logging is enabled and you can view the logs in the GUI? https://protection.office.com.

**Comment by CallMeBruce** [comment]
Hi, @jrentenaar. Thanks for the explanation and recommendation. The strangest thing, it is now working again.

For the last 2 days ResultCount was returning 0 (even though when stepping through the code in VSCode and inspecting the return ArrayList I could see data had been returned. I kludged things a bit to bypass the result/limit check and just dump the contents returned anyway as I was in a hurry). 

I have all appropriate permissions (in fact the script was working fine and between one run and the next it suddenly started this odd behaviour with the ResultCount). I did try option 1 as a test during this period and everything there returned as 0 too.

Now, it is all working again as it was. Going to put this one down to the Wrath of the Overfiend. Again, I appreciate you getting back to me.
